portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.io.IOException;
+import java.util.*; 
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.nio.charset.StandardCharsets;
+
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+  /**
+   * Delete all the comments from datastore  
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity comment: results.asIterable()) {
+      Key commentKey = comment.getKey();
+      datastore.delete(commentKey);
+    }
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
```

Comment:
Does the redirect actually do anything here?  The request comes from a javascript `fetch()` that doesn't then do anything with the response right?  So I think in that case you'd be ok to just do `response.setStatus(HttpServletResponse.SC_OK)` to let the client know everything was ok.

I can't remember off of the top of my head if you need to also send a body or something for the response to actually be sent - it might be enough to just do `response.setStatus(HttpServletResponse.SC_OK);` on its own to send an empty response.

Reply:
No, the redirect is not need, thank you for the notice. Also, the response can be sent without a body.